File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
         int n = height.length;
        int maxArea = 0;
        int left = 0, right = n - 1;
       
        while (left < right) {
            int h = Math.min(height[left], height[right]);
            int w = right - left;
            
            maxArea = Math.max(maxArea, h * w);
            
            if (height[left] < height[right]) left++;
            else right--;
            
        }
        return maxArea;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left,right=0,len(height)-1
        ans=-1
        while left<=right:
            breadth=min(height[left],height[right])
            ans=max(ans,breadth*(right-left))
            if height[left]<height[right]:
                left+=1
            else:
                right-=1
        return ans
        